File: 240913/개발자의 능력 3/ability-of-developer-3.js
const fs = require('fs');
const input = fs.readFileSync(0).toString().trim();

const abilities = input.split(' ').map(Number);

function getCombinations(arr, selectNumber) {
    const result = [];

    if ( selectNumber === 1 ) { return arr.map((e1) => [e1]) }

    arr.forEach((fixed, index, origin) => {
        const rest = origin.slice(index + 1);
        const combinations = getCombinations(rest, selectNumber - 1);

        const attached = combinations.map((combination) => [fixed, ...combination]);
        result.push(...attached);
    })

    return result;
}

function getDiff(arr) {
    const totalSum = arr.reduce((acc, cur) => acc + cur, 0);
    let minDiff = Number.MAX_SAFE_INTEGER;

    const combinations = getCombinations(arr, 3);

    combinations.forEach(team1 => {
        const team1Sum = team1.reduce((acc, cur) => acc + cur, 0);
        const team2Sum = totalSum - team1Sum;

        const curDiff = Math.abs(team1Sum - team2Sum);

        minDiff = Math.min(minDiff, curDiff)
    })

    return minDiff;
}



const ans = getDiff(abilities);
console.log(ans)

File: 240920/개발자의 능력 3/ability-of-developer-3.py
import sys;

INT_MAX = sys.maxsize

n = 6;
arr = list(map(int, input().split()));

def get_diff(i, j, k):
    sum1 = arr[i] + arr[j] + arr[k]
    sum2 = sum(arr) - sum1
    
    return abs(sum1 - sum2)

ans = INT_MAX;

for i in range(n):
    for j in range(i + 1, n):
        for k in range(j+1, n):
            ans = min(ans, get_diff(i, j, k))

print(ans)